src/entities/DFGVariable.jl

```diff
@@ -90,14 +91,22 @@ mutable struct DFGVariable <: AbstractDFGVariable
     estimateDict::Dict{Symbol, Dict{Symbol, VariableEstimate}}
     solverDataDict::Dict{Symbol, VariableNodeData}
     smallData::Dict{String, String}
-    bigData::Any
+    bigData::Dict{Symbol, AbstractBigDataEntry}
```

Comment:
can we use Option 2 method here, i.e.:
```julia
bigData::Dict{Symbol, <: AbstractBigDataEntry}
```
https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/pull/148#issuecomment-541536774

Reply:
The whole advantage from this approach is to have an abstract container. I did some benchmarks and an abstract dict with composite types in does not seem to perform worse than a typed dict.
I'm more than open for suggestions of other ways to do it, but <: AbstractBigDataEntry will not work in this case.

The advantage is in the flexibility, the types I added for illustration and the ones I currently find useful are:
```juila
struct MongodbBigDataEntry <: AbstractBigDataEntry
    key::Symbol
    oid::NTuple{12, UInt8} #mongodb object id
    #maybe other fields such as:
    #flags::Bool ready, valid, locked, permissions
    #MIMEType::Symbol
end

struct FileBigDataEntry <: AbstractBigDataEntry
    key::Symbol
    filename::String
end
```
One can just as easily define other types as long as it has a key field, maybe in memory - stored in an dictionary:
If you don't mind it not working in the cloud.
```juila
struct MemoryBigDataEntry <: AbstractBigDataEntry
    key::Symbol
    dictionaryModule::Symbol
    dictionaryReference::Symbol
end
```

The alternatives i could think of were to standardize `BigDataEntry`, but I don't know what to put in there and you lose dispatch. Or to just keep the keys in a `Set{Symbol}` kind of way.